src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1486,37 +1489,43 @@ public void setAllowFlight(boolean flight)
 	@Deprecated
 	public void hidePlayer(@NotNull Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if(!hiddenPlayersDeprecated.contains(player.getUniqueId()))
+			hiddenPlayersDeprecated.add(player.getUniqueId());
```

Comment:
If you change `hiddenPlayersDeprecated` to a `HashSet` here (since you are already preventing duplicates), you can simplify this.

```suggestion
		hiddenPlayersDeprecated.add(player.getUniqueId());
```

Reply:
Another magic class that pushes me to read the whole Java util reference